src/lib/wallet/IndyWallet.ts

```diff
@@ -30,34 +31,28 @@ export class IndyWallet implements Wallet {
     logger.log(`Wallet opened with handle: ${this.wh}`);
   }
 
-  async initPublicDid(did: Did, seed: string) {
-    let didInfo = {};
-    try {
-      const existingVerkey = await this.keyForLocalDid(did);
-      // TODO validate that verkey is as same as the verkey generated by given seed
-      didInfo = {
-        did,
-        verkey: existingVerkey,
-      };
-    } catch (error) {
-      if (error.name === 'IndyError' && error.message === '212') {
-        // WalletItemNotFound
-        logger.log('WalletItemNotFound');
-        const [_, verkey] = await this.createDid({ did, seed });
-        didInfo = {
-          did,
-          verkey,
-        };
-      } else {
-        throw error;
-      }
-    }
+  async initAgentDid(did: Did, seed: string) {
```

Comment:
From what I understand from the previous call the agent did is used for the mediator? 

I think this doesn't necessarily needs a separate function / special handling. We create a pairwise DID for the connection with the mediator just like any connection, so I don't see the benefit of using this over the already present function.

Maybe I'm missing something?

Reply:
Updated.